File: Largest number possible - GFG/largest-number-possible.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();
});

function readLine() {
    return inputString[currentLine++];
}

function printList(res,n){
    let s="";
    for(let i=0;i<n;i++){
        s+=res[i];
        s+=" ";
    }
    console.log(s);
}


function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(;i<t;i++)
    {
        let input_ar1 = readLine().split(' ').map(x=>parseInt(x));
        let n = input_ar1[0];
        let s = input_ar1[1];
        let obj = new Solution();
        let res = obj.findLargest(n, s);
        console.log(res);
        
    }
}// } Driver Code Ends



// } Driver Code Ends


//User function Template for javascript



/**
 * @param {number} n
 * @param {number} s
 * @returns {string}
*/

class Solution{
    findLargest(n,s)
    {
        if (s > n * 9 || (n > 1 && !s)) return "-1";
        let ans = "";
        for (let i = 9; i >= 0;) {
            while (s >= i) {
                ans += i.toString();
                s -= i;
                if (ans.length === n) return ans;
            }
            i--;
        }
        return ans;
    }
}

File: Largest number possible - GFG/largest-number-possible.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution{
public:
    string findLargest(int n, int s)
    {
        if (s > n*9 or n > 1 and !s) return "-1";
        string ans = "";
        for (int i = 9; i >= 0; )
        {
            while (s >= i) {
                ans += (i+'0');
                s -= i;
                if (ans.size() == n) return ans;
            }
            i--;
        }
        return ans;
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int N, S;
        cin>>N>>S;
        
        Solution ob;
        cout<<ob.findLargest(N, S)<<"\n";
    }
    return 0;
}
// } Driver Code Ends